src/main/java/net/william278/velocitab/player/Role.java

```diff
@@ -70,8 +72,34 @@ public Optional<String> getSuffix() {
     }
 
     @NotNull
-    protected String getWeightString(int highestWeight) {
-        return Placeholder.formatSortableInt(weight, highestWeight);
+    protected String getWeightString() {
+        return compressNumber(Integer.MAX_VALUE / 4d - weight);
+    }
+
+    public String compressNumber(double number) {
+        int wholePart = (int) number;
+
+        char decimalChar = (char) ((number - wholePart) * Character.MAX_VALUE);
+
+        List<Character> charList = new ArrayList<>();
+
+        while (wholePart > 0) {
+            char digit = (char) (wholePart % Character.MAX_VALUE);
+
+            charList.add(0, digit);
+
+            wholePart /= Character.MAX_VALUE;
+        }
+
+        if (charList.isEmpty()) {
+            charList.add((char) 0);
+        }
+
+        String charString = charList.stream().map(String::valueOf).collect(Collectors.joining());
+
+        charString += decimalChar;
+
+        return charString;
```

Comment:
could we just remove a _few_ of these spaces and declare vars as `final` perhaps where we can?